### ISSUE TITLE
Inductive datatypes and pattern-matching

### ISSUE THREAD

**Issue by rokopt** [anoma/juvix#910]
# Description

This is an indexing issue for tracking (in the form of linked issues) what's incomplete or broken in the typechecking and compilation of inductive datatypes, and pattern-matching to define functions on those datatypes.  In particular we can define for each linked issue whether it's a requirement for the [Stygarchus release](https://github.com/anoma/juvix/issues/890) or whether it will be deferred until after that.

# Blocks

# Blocked by

# See also

- #890

**Comment by cwgoes** [comment]
I suggest that we focus first on the part of the pipeline prior to backends, and in parallel get the LLVM backend to where ti can compile basic datatypes & pattern-matching.

**Comment by rokopt** [comment]
Regarding pattern-matching, see #492 , and in particular https://github.com/anoma/juvix/issues/492#issuecomment-719644992.

**Comment by rokopt** [comment]
See #985 for a proposed start (product/coproduct types), along with a path towards a full-featured implementation (with recursive and dependent types, for example, and guards on pattern-matches).
